packages/cyberstorm-forms/src/components/FormSelect.tsx

```diff
@@ -0,0 +1,126 @@
+"use client";
+
+import { z, ZodObject } from "zod";
+import { ZodRawShape } from "zod/lib/types";
+import { Path, useController } from "react-hook-form";
+import styles from "./FormSelect.module.css";
+import * as Select from "@radix-ui/react-select";
+import React, { PropsWithChildren } from "react";
+import { Icon } from "@thunderstore/cyberstorm";
+import { faCaretDown, faCheckCircle } from "@fortawesome/pro-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+export type FormSelectProps<
+  Schema extends ZodObject<Z>,
+  Z extends ZodRawShape
+> = {
+  // The schema is required to allow TS to infer valid values for the name field
+  schema: Schema;
+  name: Path<z.infer<Schema>>;
+  options: { value: string; label: string }[];
+  placeholder?: string;
+  defaultValue?: string;
+  ariaLabel?: string;
+};
+export function FormSelect<Schema extends ZodObject<Z>, Z extends ZodRawShape>({
+  name,
+  defaultValue,
+  options,
+  ariaLabel,
+}: FormSelectProps<Schema, Z>) {
+  const {
+    field,
+    fieldState: { error },
+    formState: { isSubmitting, disabled },
+  } = useController({ name });
+
+  if (defaultValue) {
+    React.useEffect(() => {
+      field.onChange(defaultValue);
+    }, [defaultValue]);
+  }
+
+  return (
+    <>
```

Comment:
Per my earlier review, this should probably be `div` to keep the field and error together. Don't know which approach you ended up taking in the earlier PR.

Reply:
Ended up going with those "no element" way, so that the elements can be easily controlled in the form level.

If you have some example case where having these wrapped in a div would be useful; let me know.

Reply:
Well I don't really have one, I guess it depends on the CSS used in the forms themselves. Just keep an eye out: if you end up wrapping all the elements in div, it should probably be done in the component rather than repeated on the outside of it.